### PR TITLE
Update plugins / themes via CRON

### PR DESCRIPTION
**Pull request by khromov**
This PR stems from the discussion in #242 

By introducing the `$force` option to get_plugin_meta() and `$force_meta_update` to the Plugin and Theme class constructors, we no longer make external calls on a regular (admin) page load.

By introducing a number of hooks in the new `background_update()` function, we make sure that the plugin metadata will attempt to refresh on the following conditions:

* Visiting plugin page
* Visiting update page
* When`wp_update_plugins` and `wp_update_themes` hooks run. (These run on the CRON jobs with the same name. WordPress automatically sets up these jobs to run every 12 hours.)

Basically, on the above conditions, the plugin will work as it did before. On all other pages it will *not* attempt to grab data remotely. (That is what the CRON job is for!)

You can try this out by installing the Crontrol plugin and manually running the `wp_update_plugins` and `wp_update_themes` CRON jobs.

I made this as a minimal working proof of concept, using the existing code structure. I haven't tested branch switching but seeing as the changes to the code are minimal, it should work as before. Any feedback welcome.

### CODE REVIEW

**Comment by afragen** [comment]
Shouldn't there be a corresponding change to https://github.com/afragen/github-updater/pull/270/files#diff-826f6032956806670f536a25ad599654R90 adding the `$this->force_meta_update` boolean?

**Comment by afragen** [comment]
I'm creating a new branch for this PR so I can test before complete merge. Can you adjust the PR to `wp-cron` branch?

**Comment by khromov** [comment]
GitHub won't let you change branches on a PR but I created a new one here: #272
